great_expectations/core/batch_config.py

```diff
@@ -0,0 +1,6 @@
+from typing_extensions import TypeAlias
+
+from great_expectations.datasource.fluent.sql_datasource import QueryAsset
+
+# TODO: fix this type
+BatchConfig: TypeAlias = QueryAsset
```

Comment:
The type here should be [DataAsset](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/datasource/fluent/interfaces.py#L161) which is the superclass of all data assets and has [`build_batch_request` as part of it's interface ](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/datasource/fluent/interfaces.py#L214).

I think we want this as a new type that composes in the DataAsset from which it is derived. I'm worried the API for the config object will be different than that of an asset. For example, splitters and sorters will probably need to be removed from assets and be made part of the `BatchConfig`.